### ISSUE TITLE
lesson 2 - Read cyclical learning rates for training neural networks

### ISSUE THREAD

**Issue by datalass1** [datalass1/fastai#9]
https://arxiv.org/pdf/1506.01186.pdf

**Comment by datalass1** [comment]
The essence of this learning rate policy comes from the
observation that increasing the learning rate might have a
short term negative effect and yet achieve a longer term beneficial
effect. 

This observation leads to the idea of letting the
learning rate vary within a range of values rather than adopting
a stepwise fixed or exponentially decreasing value. That
is, one sets minimum and maximum boundaries and the
learning rate cyclically varies between these bounds. 

The paper is focused on the triangular
learning rate policy, all others were equivalent and the triangular is simplest. 

The length of a cycle and the input parameter stepsize
can be easily computed from the number of iterations in
an epoch. **An epoch is calculated by dividing the number
of training images by the batchsize used. For example,
CIFAR-10 has 50, 000 training images and the batchsize is
100 so an epoch = 50, 000/100 = 500 iterations**

**How can one estimate reasonable minimum and
maximum boundary values?**
A “LR range test”; run your
model for several epochs while letting the learning rate increase
linearly between low and high LR values. This test
is enormously valuable whenever you are facing a new architecture
or dataset.
